bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/graphics/Image.java

```diff
@@ -1837,5 +1839,73 @@ public static void drawAtSize(GC gc, ImageData imageData, int width, int height)
 	});
 }
 
+void executeOnImageAtSizeBestFittingSize(Consumer<Image> imageAtBestFittingSizeConsumer, int destWidth, int destHeight) {
+	Optional<Image> imageAtSize = cachedImageAtSize.refresh(destWidth, destHeight);
+	imageAtBestFittingSizeConsumer.accept(imageAtSize.orElse(this));
+}
+
+private CachedImageAtSize cachedImageAtSize = new CachedImageAtSize();
+
+private class CachedImageAtSize {
+	private Image image;
+
+	public void destroy() {
+		if (image != null) {
+			image.dispose();
+			image = null;
+		}
+	}
+
+	private Optional<Image> refresh(int destWidth, int destHeight) {
+		int scaledWidth = pointToPixel(destWidth, DPIUtil.getDeviceZoom());
+		int scaledHeight = pointToPixel(destHeight, DPIUtil.getDeviceZoom());
+		if (isReusable(scaledWidth, scaledHeight)) {
+			return Optional.of(image);
+		} else {
+			destroy();
+			Optional<Image> imageAtSize = loadImageAtSize(scaledWidth, scaledHeight);
+			image = imageAtSize.orElse(null);
+			return imageAtSize;
+		}
+	}
+
+	private boolean isReusable(int width, int height) {
+		return image != null && image.height == height && image.width == width;
+	}
+
+	private Optional<Image> loadImageAtSize(int destWidth, int destHeight) {
+		Optional<ImageData> imageData = loadImageDataAtExactSize(destWidth, destHeight);
+		if (imageData.isEmpty()) {
+			return Optional.empty();
+		}
+		Image image = new Image(device, imageData.get());
+		NSBitmapImageRep representation = image.getRepresentation(100);
```

Comment:
Shouldnt the image.getRepresentation be inside the if condition? also not very clear to me why do we get and remove representation at 100.

Reply:
Yes, it can be inside the if block. I fixed that with the latest commit.
We create the image based on image data in the line before (1881), which effectively creates an image with a 100% representation (out of the given image data). For that reason, we retrieve the representation at that same zoom. We then use the `createRepFromSourceAndApplyFlag` to apply the style flag on that representation (which adds another representation with that styling to the image), so that we need to remove the original (unstyled) representation afterwards.